schema2Markdown.js

```diff
@@ -0,0 +1,255 @@
+            markdownContent += '\n```\n\n';
+        }
+        const url = filePath.replace("schemas/", `https://github.com/finos/FDC3/tree/master/schemas/`);
+        markdownContent += `Generated from ${url}`;
+        
+    }
+
+    const frontMatter = generateFrontMatter(objectName, schema.description);
+
+    const outputFileName = `./website/versioned_docs/version-${version}/${schemaFolderName}/schemas/${title.replace(/\s+/g, '')}.md`;
+
+    fse.outputFileSync(outputFileName, `---\n${yaml.dump(frontMatter)}\n---\n\n${markdownContent}`);
+
+    // objectName must not contain any spaces
+    if (objectName != null) {
+        return schemaFolderName + '/schemas/' + objectName.replace(/\s+/g, '');
+    }
+}
+
+function escape(text) {
+    return text.replace(/>/g, '\\>');
+}
+
+function generateFrontMatter(title, description) {
+    return {
+        title: `${title} Schema`,
+        description: description,
+        sidebar_label: title + ' Schema',
+    };
+}
+
+function processSchemaFile(schemaFile, schemaFolderName, version) {
+    const schemaData = fse.readJSONSync(schemaFile);
+
+    // if there is allOf, then it is an object
+    const allOfArray = schemaData.allOf;
+    let sidebarItems = [];
+    if (Array.isArray(allOfArray) && allOfArray.length > 0) {
+        sidebarItems.push(generateObjectMD(schemaData, null, schemaFolderName, schemaFile, version));
+    }
+    if (schemaData.definitions) {
+        for (const [objectName, objectDetails] of Object.entries(schemaData.definitions)) {
+            sidebarItems.push(generateObjectMD(objectDetails, objectName, schemaFolderName, schemaFile, version));
+        }
+    }
+
+    return sidebarItems;
+}
+
+function parseSchemaFolder(schemaFolderName, version) {
+    // Read all files in the schema folder
+    const schemaFiles = fse.readdirSync("./schemas/"+schemaFolderName)
+    .filter(file => file.endsWith('.json'))
+    .map(file => path.join("./schemas/"+schemaFolderName, file));
+
+    // Process each schema file
+    let sidebarItems = [];
+    for (const schemaFile of schemaFiles) {
+        sidebarItems.push(processSchemaFile(schemaFile, schemaFolderName, version));
+    }
+
+    // filter out null values
+    return sidebarItems.flat().filter(item => item);
+}
+
+function main() {
+
+    const versions = fse.readdirSync('./website/versioned_docs').map(version => version.split('-')[1]);
+
+    versions.forEach(version => {
+
+        let sidebarObject = fse.readJsonSync(`./website/versioned_sidebars/version-${version}-sidebars.json`)    
+
+        let sidebarContextObject = {
+            "type": "category",
+            "label": "Context Schemas Part",
+            "items": []
+        }
+
+        let sidebarApiObject = {
+            "type": "category",
+            "label": "API Schemas Part",
+            "items": []
+        }
+
+        sidebarApiObject.items = parseSchemaFolder('api', version);
```

Comment:
We don't need a reference generating for this yet - when we do it'll probably be incorporated into the FDC3 protocol spec and handled slightly differently

```suggestion
```